htdocs/xoops_lib/modules/protector/class/protector.php

```diff
@@ -445,7 +445,7 @@ public function get_bad_ips($with_jailed_time = false)
      */
     public static function get_filepath4badips()
     {
-        return XOOPS_VAR_PATH . '/protector/badips' . substr(md5(XOOPS_ROOT_PATH . XOOPS_DB_USER . XOOPS_DB_PREFIX), 0, 6);
+        return XOOPS_VAR_PATH . '/protector/configcache' . substr(md5(XOOPS_ROOT_PATH . XOOPS_DB_USER . XOOPS_DB_PREFIX), 0, 6);
```

Comment:
This does not make any sense. The file name form is badipsXXXXXX. The configcache is a totally separate file, and its name is returned by \Protector::get_filepath4confighcache().

What is the motivation for the change?

Reply:
Just because it found something doesn't mean it is the correct file ;)

configcache is generated automatically when it is not found or when the preferences are changed.

badips and group1ips are generated when the lists at the top of protect center are updated. Also, badips is updated when a ban is done, adding the ip and possibly duration.

I'm not sure why you are seeing an error; protector uses the error control operator (@) when it accesses it, so that hasn't normally logged anything.

Perhaps, protector should write its own empty files when it is installed, but this change is definitely wrong.

Reply:
It was a fresh installation on PHP 7.3 RC4, and to be precise, it was a warning or notice. Because of the issues I had with losing the connection, I was trying to eliminate all possible issues, and after this change, it notice/warning went away.

Reply:
I also tested on 7.3 RC4 and still did not see any errors. Could there be an issue with the WAMP build?

Reply:
Question: did that show in the XOOPS logger, or did you see it in a debugger?

Reply:
It could be an issue with the WAMP, or just with the installation being on Windows. 
It was in Console view of the Debug window in PhpStorm

Reply:
OK, that makes sense. The error does occur, so you could see it in debugger, but the error would never show to the user, as it is silenced with "@".  (That actually works by turning off the error reporting temporarily, so that error_reporting() returns 0. Error handlers, like \XoopsLogger::handleError(), are supposed to check that.)